source/frontend/src/models/generate/acquisition/GenerateCompensationPayee.ts

```diff
@@ -15,46 +17,54 @@ export class Api_GenerateCompensationPayee {
     compensation: ApiGen_Concepts_CompensationRequisition | null,
     financialActivities: ApiGen_Concepts_CompensationFinancial[] | [],
   ) {
-    this.gst_number = compensation?.gstNumber ? compensation.gstNumber ?? '' : '';
+    this.gst_number = compensation?.gstNumber ?? '';
 
     if (compensation?.acquisitionOwner) {
       this.name = formatNames([
-        compensation.acquisitionOwner.givenName,
-        compensation.acquisitionOwner.lastNameAndCorpName,
+        compensation?.acquisitionOwner.givenName,
+        compensation?.acquisitionOwner.lastNameAndCorpName,
       ]);
     } else if (compensation?.interestHolder) {
-      if (compensation.interestHolder.person) {
+      if (compensation?.interestHolder.person) {
         this.name = formatNames([
-          compensation.interestHolder.person.firstName,
-          compensation.interestHolder.person.surname,
+          compensation?.interestHolder.person.firstName,
+          compensation?.interestHolder.person.surname,
         ]);
       } else {
-        this.name = compensation.interestHolder.organization?.name ?? '';
+        this.name = compensation?.interestHolder.organization?.name ?? '';
       }
     } else if (compensation?.acquisitionFileTeam) {
-      if (compensation.acquisitionFileTeam.person) {
+      if (compensation?.acquisitionFileTeam.person) {
         this.name = formatNames([
-          compensation.acquisitionFileTeam.person.firstName,
-          compensation.acquisitionFileTeam.person.surname,
+          compensation?.acquisitionFileTeam.person.firstName,
+          compensation?.acquisitionFileTeam.person.surname,
         ]);
       } else {
-        this.name = compensation.acquisitionFileTeam.organization?.name ?? '';
+        this.name = compensation?.acquisitionFileTeam.organization?.name ?? '';
       }
     } else if (compensation?.legacyPayee) {
-      this.name = compensation.legacyPayee ?? '';
+      this.name = compensation?.legacyPayee ?? '';
+    } else if (compensation?.compReqLeaseStakeholder?.length > 0) {
+      const stakeHolder: ApiGen_Concepts_LeaseStakeholder =
+        compensation?.compReqLeaseStakeholder[0].leaseStakeholder;
+      if (stakeHolder.lessorType.id === ApiGen_CodeTypes_LessorTypes.ORG) {
+        this.name = stakeHolder.organization.name ?? '';
```

Comment:
Just to be safe - I would apply a nullcheck here for organization like the rest of the code in this file:

```ts
this.name = stakeholder.organization?.name ?? '';
```

Reply:
Updated